### PR TITLE
Process API lecture

### PR DESCRIPTION
**Pull request by pbrinkmeier**
On the one hand, i finally finished the script for the lecture process API.

On the other hand, I made some typographic improvements and would like us to use this document as a template for the future. The improvements include:
- Correct use of &ldquo;quotation marks&rdquo; instead of inch marks (`&ldquo;`, `&rdquo;`)
- Correct use of en&ndash;dash and also &mdash; em&ndash;dash instead of hyphens (`&ndash;`, `&mdash;`)
- Differentiation:
   - italics are used for *emphasis*
   - bold text designates **important concepts**
- **Do not**: include punctation in emphasis, e.g. **stack pointer** (SP) instead of **stack pointer (SP)**
- Correct use of apostrophes instead of single ticks (`&rsquo;`)
- Put each sentence in a paragraph in its own line, whitespace is collapsed anyway and it makes for easier editing
- For headings, always use heading. Never use bold text, it&rsquo;s not semantic.

### CODE REVIEW

**Comment by pbrinkmeier** [comment]
I need to reformat the ordered lists, my markdown editor renders them differently. Also, I need to re&endash;generate the TOC.

File: betriebssysteme/04-process-api.md [hunk 1]
```diff
@@ -18,31 +18,42 @@
...
 
-The _stack pointer (SP)_ holds the address of the top of the stack while the stack grows _downwards_. The SP points to the last allocated word ("pre-decremented stack pointer").
+- **jmp**: continue execution at given address  
+- **je**: &ldquo;jump equal&rdquo; (jump if condition is true)  
+- **jz**: &ldquo;jump zero&rdquo; (jump if operand is zero)  
+- **call**: jump to a function (subroutine) by pushing current code location to stack  
+- **return**:  return from subroutine (to return address on the stack)
 
-_push_ makes room for values on the stack by decrementing the SP and the new element. _pop_ cleans up values from the stack by incrementing the SP (removed data is not overwritten).
+#### Stack
```
Comment (luk3b) [inline betriebssysteme/04-process-api.md]: I'd keep the (x86) comment, stacks do not work equally on every cpu architecture.

File: betriebssysteme/04-process-api.md [hunk 2]
```diff
@@ -58,13 +58,67 @@ It may specify executable/object file layout, calling conventions, stack alignme
...
+- CPU registers (limit depends on how many registers there are)
+- Main memory (heap or stack) for strings or if the registers are not enough
+
+Arguments may also just all be transferred via main memory.
+
+A **return code** needs to be returned to the application. Usually, negative numbers denote an error condition, positive numbers and zero denote success. Return codes are usually stored in the A and D registers.
+
+### System call handler
+
+When an application &ldquo;traps&rdquo; into the Kernel, the System call handler
```
Comment (luk3b) [inline betriebssysteme/04-process-api.md]: I don't really like the idea of using the HTML unicode variant instead of normal quotation marks

**Comment by pbrinkmeier** [comment]
So besser? :)

**Pull request merged**